[SPARK-19564][SS][KAFKA] KafkaOffsetReader's consumers should not be in the same group

lw-lin opened the pull request:
## What changes were proposed in this pull request?

In `KafkaOffsetReader`, when error occurs, we abort the existing consumer and create a new consumer. In our current implementation, the first consumer and the second consumer would be in the same group, which violates our intention of the two consumers not being in the same group.

The cause is that, in our current implementation, the first consumer is created before `groupId` and `nextId` are initialized in the constructor. Then even if `groupId` and `nextId` are increased during the creation of that first consumer, `groupId` and `nextId` would still be initialized to default values in the constructor.

We should make sure that `groupId` and `nextId` are initialized before any consumer is created.

## How was this patch tested?

Ran 100 times of `KafkaSourceSuite`; all passed

lw-lin committed:
Fix

SparkQA commented:
**[Test build #72772 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/72772/testReport)** for PR 16900 at commit [`f0225a0`](https://github.com/apache/spark/commit/f0225a0a5967cd554a54f4c1b60d8542a5e88d77).

lw-lin commented:
Let's start by running only `KafkaSourceSuite` for 1000 times

SparkQA commented:
**[Test build #72773 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/72773/testReport)** for PR 16900 at commit [`b0e549d`](https://github.com/apache/spark/commit/b0e549dddb38cbb8efde4526fbeb6ce67d7650b7).

SparkQA commented:
**[Test build #72773 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/72773/testReport)** for PR 16900 at commit [`b0e549d`](https://github.com/apache/spark/commit/b0e549dddb38cbb8efde4526fbeb6ce67d7650b7).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/72773/
Test PASSed.

lw-lin commented:
ok, all 100 times of `KafkaSourceSuite` passed.

let's now do a full pass of testing.

SparkQA commented:
**[Test build #72778 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/72778/testReport)** for PR 16900 at commit [`32e92ce`](https://github.com/apache/spark/commit/32e92ce76f9f51b10fede69196e22fe62602ec3f).

SparkQA commented:
**[Test build #72778 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/72778/testReport)** for PR 16900 at commit [`32e92ce`](https://github.com/apache/spark/commit/32e92ce76f9f51b10fede69196e22fe62602ec3f).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/72778/
Test PASSed.

lw-lin committed:
Run 100 times

# This is the 1st commit message:

Run 100 times

# This is the commit message #2:

Revert "Run 100 times"

SparkQA commented:
**[Test build #72783 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/72783/testReport)** for PR 16900 at commit [`0f19af3`](https://github.com/apache/spark/commit/0f19af3b6b2fdc8f527f952bbf65681755c3f6b4).

SparkQA commented:
**[Test build #72783 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/72783/testReport)** for PR 16900 at commit [`0f19af3`](https://github.com/apache/spark/commit/0f19af3b6b2fdc8f527f952bbf65681755c3f6b4).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/72783/
Test PASSed.

SparkQA commented:
**[Test build #72772 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/72772/testReport)** for PR 16900 at commit [`f0225a0`](https://github.com/apache/spark/commit/f0225a0a5967cd554a54f4c1b60d8542a5e88d77).
 * This patch **fails from timeout after a configured wait of \`250m\`**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/72772/
Test FAILed.
